### ISSUE TITLE
Routing does not consider width and maxwidth even included in routing.xml 

### ISSUE THREAD

**Issue by TomSH75** [osmandapp/Osmand#7555]
**OsmAnd version**: 3.4.8
Do you use the offline maps offered within the OsmAnd app for download, or online (tile / raster) maps?
Offline maps offered within OsmAnd app

**Description of issue:**
I modified the routing.xml file in order to consider the width and maxwidth of a road. I would like to avoid with my motor home too narrow roads & bridges...

Modifications I did to routing.xml but the routing does not consider the limitations (width of 5m and maxwidth of 2.5m when set. Limitations of maxweigth and maxheight do work. 

Attached my modified **routing.xml**:
[routing.xml.txt](https://github.com/osmandapp/Osmand/files/3612839/routing.xml.txt)

This part works properly:
`<parameter id="width_max" name="WidthMax" description="Maximum vehicle width to be permitted on the route" type="numeric" values="0,1.49,1.99,2.29,2.49" valueDescriptions="-,1.5m,2m,2.3m,2.5m"/>
<parameter id="width_street" name="WidthStreet" description="Minimum width of street to pass." type="numeric" values="0,1.49,1.99,2.29,4.99,10" valueDescriptions="-,1.5m,2m,2.3m,5m,10m"/>`

Changes to the routing engine:
Access section:
`<select value="-1" t="maxwidth">
    <gt value1=":width_max" value2="$maxwidth" type="length"/>
   </select>
<select value="-1" t="width">
      <gt value1=":width_street" value2="$width" type="length"/>
   </select>`

Obstacle section:
`<select value="-1" t="maxwidth">
    <gt value1=":width_max" value2="$maxwidth" type="length"/>
   </select>
<select value="-1" t="width">
     <gt value1=":width_street" value2="$width" type="length"/>
   </select>`
Routing used: OsmAnd's in-app offline routing

Car routing profile chosen from attached routing.xml

**maxwidth topic:**
_wrong routing:_
Bridge https://www.openstreetmap.org/way/26936358 has a limit of 2.4m
![wrong routing_max width topic](https://user-images.githubusercontent.com/55321799/64909956-42b43400-d712-11e9-89f0-f910d8637443.jpeg)
_expected routing:_
![routing expected_max width topic](https://user-images.githubusercontent.com/55321799/64909954-42b43400-d712-11e9-824f-1efb28922510.jpeg)

**Width topic:**
_wrong routing:_
Street https://www.openstreetmap.org/way/344858724 has a width of 4m
![wrong routing_width topic](https://user-images.githubusercontent.com/55321799/64910017-f1f10b00-d712-11e9-937d-501d2f371673.jpeg)
_expected routing:_
![routing expected_width topic](https://user-images.githubusercontent.com/55321799/64910020-fd443680-d712-11e9-9224-1d62dfee8e21.jpeg)

**Comment by MadWasp79** [comment]
@TomSH75 we reviewed your issue and decided that this feature (check for the maximum width of roads) will be added to our default profiles in the next app update. 
For now, it is not possible to implement, even with custom profiles, since we didn't include this tag ("maxwidth") to our maps. This too will be fixed in the next monthly maps update.

**Comment by TomSH75** [comment]
Hello Denis,
you write "check for the maximum width of roads" and "maxwidth" tag. Do 
you plan to integrate only the maxwidth tag or the width AND maxwidth 
tag?

Actually the more benefit for routing I expect from the width tag. The 
maxwidth sign according to my experience very seldom is used. However 
narrow streets which trucks / mobil homes and caravans would like to 
avoid you can find everywhere.

Best regards,

Tom

------ Originalnachricht ------
Von: "Denis Kolomiets" <notifications@github.com>
An: "osmandapp/Osmand" <Osmand@noreply.github.com>
Cc: "Tom Schulz-Heise" <tom@schulz-heise.de>; "Mention" 
<mention@noreply.github.com>
Gesendet: 07.10.2019 13:48:28
Betreff: Re: [osmandapp/Osmand] Routing does not consider width and 
maxwidth even included in routing.xml (#7555)

**Comment by MadWasp79** [comment]
Both. There is "only" 28k of maxwidth tags in osm maps and almost 2 millions of width tags.

**Comment by TomSH75** [comment]
I tested it with the latest version of OSMand and updated maps. Works great! Attached my latest routing.xml. Please note that the routing.xml has to be put into the /routing/ directory (e.g. sdcard/Android/data/net.osmand.plus/files/routing/)
[routing.xml.txt](https://github.com/osmandapp/Osmand/files/3878619/routing.xml.txt)

**Comment by NickSlash** [comment]
I was/am having issues with maxwidth not restricting my routes too. There are some nodes tagged as `traffic_calming` with a `maxwidth` and there specified in feet and inches (as sign posted).

The underlying way also had `maxwidth` specified, again in feet and inches.

https://www.openstreetmap.org/way/126358294

On a custom map, I tried converting the `traffic_calming` node to metric but this had no effect on routing. I also tried converting the `maxwidth` value on the underlying way to meters and it seems to have fixed my problem.

I'm fairly new to using mapcreator so I may just have a configuration issue but I think there may be an issue with meters vs feet and inches?
